packages/headless/src/controllers/generated-answer/headless-generated-answer.test.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  dislikeGeneratedAnswer,
+  likeGeneratedAnswer,
+  resetAnswer,
+  streamAnswer,
+} from '../../features/generated-answer/generated-answer-actions';
+import {
+  logDislikeGeneratedAnswer,
+  logLikeGeneratedAnswer,
+  logOpenGeneratedAnswerSource,
+} from '../../features/generated-answer/generated-answer-analytics-actions';
+import {generatedAnswerReducer} from '../../features/generated-answer/generated-answer-slice';
+import {getGeneratedAnswerInitialState} from '../../features/generated-answer/generated-answer-state';
+import {executeSearch} from '../../features/search/search-actions';
+import {buildMockCitation} from '../../test/mock-citation';
+import {
+  buildMockSearchAppEngine,
+  MockSearchEngine,
+} from '../../test/mock-engine';
+import {
+  buildGeneratedAnswer,
+  GeneratedAnswer,
+} from './headless-generated-answer';
+
+describe('generated answer', () => {
+  let generatedAnswer: GeneratedAnswer;
+  let engine: MockSearchEngine;
+
+  function initGeneratedAnswer() {
+    generatedAnswer = buildGeneratedAnswer(engine);
+  }
+
+  function findAction(actionType: string) {
+    return engine.actions.find((a) => a.type === actionType);
+  }
+
+  beforeEach(() => {
+    engine = buildMockSearchAppEngine();
+    initGeneratedAnswer();
+  });
+
+  it('it adds the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      generatedAnswer: generatedAnswerReducer,
+    });
+  });
+
+  it('should return the state', () => {
+    expect(generatedAnswer.state).toEqual(getGeneratedAnswerInitialState());
+  });
+
+  it('should subscribe to state updates', () => {
+    expect(engine.subscribe).toHaveBeenCalledTimes(1);
+  });
+
+  it('#retry dispatches #executeSearch', () => {
+    generatedAnswer.retry();
+    const action = engine.findAsyncAction(executeSearch.pending);
+
+    expect(action).toBeTruthy();
```

Comment:
can we assert falsy before retry()?

Reply:
sure thing 👍

Reply:
But I don't think we do that in any other tests asserting that an action can be found

Reply:
Yes I noticed the other tests and thought the same. Since these are unit tests with state reset before each test, guess it is fine to leave them as they are.